pick.c

```diff
@@ -928,16 +930,20 @@ get_key(char *buf, size_t size, size_t *nread)
 		KEY(RIGHT,	"\006"),
 		KEY(UNKNOWN,	NULL),
 	};
-	int	c, i;
+	static unsigned char	buf[8];
```

Comment:
Out of curiosity: What are the pros of scoping this static variable to `get_key` and letting `*key` point to it rather than declaring it outside `get_key` and letting `selected_choice` refer directly to it? Making it possible to declare it `const`?

The size of `buf` seems to have increased from `6` to `8`. What was the reason for doing that?

Reply:
> Out of curiosity: What are the pros of scoping this static variable to `get_key` and letting `*key` point to it rather than declaring it outside `get_key` and letting `selected_choice` refer directly to it? Making it possible to declare it `const`?

Less dereferencing going on in `get_key` and if we where to declare
`buf` as `unsigned` in `selected_choice` it would require lots of
casting while passing it to various functions. I find the `get_key` code
a bit easier to follow after applying this diff, but that's just my
opinion.

> The size of `buf` seems to have increased from `6` to `8`. What was the reason for doing that?

No real reason, other than 8 being a power of 2.